plugins/awards-backend/src/service/router.ts

```diff
@@ -2,28 +2,69 @@
  * Copyright SeatGeek
  * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
  */
-import { PluginDatabaseManager, errorHandler } from '@backstage/backend-common';
+import {
+  PluginDatabaseManager,
+  TokenManager,
+  errorHandler,
+} from '@backstage/backend-common';
+import { DiscoveryService } from '@backstage/backend-plugin-api';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Config } from '@backstage/config';
 import { AuthenticationError } from '@backstage/errors';
 import { IdentityApi } from '@backstage/plugin-auth-node';
 import express from 'express';
 import Router from 'express-promise-router';
 import { Logger } from 'winston';
 import { Awards } from '../awards';
 import { DatabaseAwardsStore } from '../database/awards';
+import {
+  NotificationsGateway,
+  NullNotificationGateway,
+  SlackNotificationsGateway,
+} from '../notifications/notifications';
 
 export interface RouterOptions {
   identity: IdentityApi;
   database: PluginDatabaseManager;
   logger: Logger;
+  // todo: make required in next breaking change
+  config?: Config;
```

Comment:
or, of course, just follow semver and be ok doing a major version release

Reply:
> ideally we would have started major version zero but the semantic release tool explicitly doesn’t allow this

I was initially disappointed by this, but their reasoning does make sense.  I'd strongly prefer that we avoid intentional BC breaks, since breaking BC tends to break trust. I think we should follow semver (and do a major version release), optionally with some pre-releases first if we need time to test and refine the changes before cutting that stable release.

Reply:
+1 to following semver.  It's more important for releases to be reliable (and changes expected) than for major releases to include a lot of changes.

Reply:
ok this all sounds good. once these branches are merged i'm going to set up a prelease flow and won't cut a proper release until we've run locally a bit

Reply:
hm, might just try to cut the release after shipping these changes and save prereleasing for later. seems like it'll be a good amount of work to get right